Data_Structures_Answers.md

```diff
@@ -1,18 +1,22 @@
 Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
+Only one if statment O(1)
 
 2. What is the space complexity of your ring buffer's `append` function?
+    O(n)
 
 3. What is the runtime complexity of your ring buffer's `get` method?
+    One loop and if staitments O(n)
 
 4. What is the space complexity of your ring buffer's `get` method?
-
+    O(n)
 
 5. What is the runtime complexity of the provided code in `names.py`?
-
+    O(n)
 6. What is the space complexity of the provided code in `names.py`?
-
+    O(n)
 7. What is the runtime complexity of your optimized code in `names.py`?
-
+     0.005 on my MacBook, O(n)
```

Comment:
the actual solution would be O(n^2) because of the nested for loops.

Reply:
fixed thank you for comment